app/Http/Controllers/TaskCategoriesController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Auth;
+use Illuminate\Http\Request;
+use App\Models\TaskCategory;
+use Illuminate\Support\Facades\Validator;
+use App\Http\Requests\TaskCategoryStoreRequest;
+
+class TaskCategoriesController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categoryList = TaskCategory::where('user_id', Auth::user()->id)->get();
+        return view('task-category.index', compact('categoryList'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('task-category.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TaskCategoryStoreRequest $request)
+    {
+        $taskCategory = new TaskCategory;
+        $taskCategory->user_id = Auth::user()->id;
+        $taskCategory->name = $request->name;
+        $taskCategory->save();
+
+        return redirect()->route('taskcategory.index')->with('success', 'Data Added successfully!');
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+
+    public function edit($id)
```

Comment:
Use route model binding as mentioned earlier as well.

Reply:
I've used route model binding in TaskController as you can check but not working in TaskCategoriesController.

Reply:
It should work. @amit-ranium help Swarup and see why route model binding is not working.

Reply:
Sir, Found issue and now its working. Variable name used for route model binding should be same as the route name.

Reply:
Yes, correct